modules/seqexec/server/src/main/scala/seqexec/server/gnirs/GnirsControllerEpics.scala

```diff
@@ -103,17 +92,25 @@ object GnirsControllerEpics {
   }
 
   implicit val wavelEncoder: EncodeEpicsValue[Wavelength, Double] = EncodeEpicsValue(_.toNanometers)
+}
+
+object GnirsControllerEpics extends GnirsEncoders {
+  private val Log = getLogger
+
+  private def epicsSys = GnirsEpics.instance
+  private def ccCmd = epicsSys.configCCCmd
+  private def dcCmd = epicsSys.configDCCmd
 
-  private def setAcquisitionMirror(mode: Mode): List[SeqAction[Unit]] = {
+  private def setAcquisitionMirror(mode: Mode): IO[Option[IO[Unit]]] = {
     val v = mode match {
       case Acquisition => "In"
       case _           => "Out"
     }
 
-    smartSetParam(v, epicsSys.acqMirror.map(removePartName), ccCmd.setAcqMirror(v))
+    smartSetParamF(v, Nested(epicsSys.acqMirror).map(removePartName).value, ccCmd.setAcqMirror(v))
   }
 
-  private def setGrating(s: Spectrography, c: Camera): List[SeqAction[Unit]] = {
+  private def setGrating(s: Spectrography, c: Camera): List[IO[Option[IO[Unit]]]] = {
```

Comment:
In these case an the ones below, couldn't `IO[List[IO[Unit]]]` be used instead of `List[IO[Option[IO[Unit]]]]`.

Reply:
I've tried to implement this on a different way but the issue I found is that I need to know if there are elements on the list. If there are no elements on the list I shouldn't execute the `after` part on https://github.com/gemini-hlsw/ocs3/blob/develop/modules/seqexec/server/src/main/scala/seqexec/server/EpicsUtil.scala#L426
That is needed to avoid calling `post` if there are no changes.
I can still look for a better way to do this. I agree it is hard to follow